src/storage/secondary/column/primitive_column_builder.rs

```diff
@@ -121,7 +121,7 @@ pub fn append_one_by_one<'a, A: Array>(
 impl<T: PrimitiveFixedWidthEncode> ColumnBuilder<T::ArrayType> for PrimitiveColumnBuilder<T> {
     fn append(&mut self, array: &T::ArrayType) {
         let mut iter = array.iter().peekable();
-
+        let mut first_flag = true;
```

Comment:
first_flag should be an inner state of `ColumnBuilder` instead of a local variable. One append doesn't necessarily produce a block.

I think a better way is to store the first_key in the struct on switching builder (when `self.current_builder.is_none()`)?

Reply:
> first_flag should be an inner state of `ColumnBuilder` instead of a local variable. One append doesn't necessarily produce a block.
> 
> I think a better way is to store the first_key in the struct on switching builder (when `self.current_builder.is_none()`)?

Yes, the condition `self.current_builder.is_none()` can be reused.